components/ILIAS/Contact/classes/class.ilMailSearchObjectGUI.php

```diff
@@ -136,137 +137,48 @@ protected function addPermission(array $a_obj_ids): void
-                );
-            }
-
-            if ($usr_ids !== []) {
-                $this->addPermission($usr_ids);
-            } else {
-                $this->tpl->setOnScreenMessage('info', $this->lng->txt('mail_select_one_entry'));
-                $this->showMembers();
-            }
+        if ($obj_ids !== []) {
+            $this->addPermission($obj_ids);
         } else {
+            $this->tpl->setOnScreenMessage('info', $this->lng->txt('mail_select_course'));
             $this->showMyObjects();
         }
     }
 
-    protected function mail(): void
+    private function shareMembers(): void
     {
-        $view = '';
-        if ($this->http->wrapper()->query()->has('view')) {
-            $view = $this->http->wrapper()->query()->retrieve('view', $this->refinery->kindlyTo()->string());
-        }
-
-        if ($view === 'myobjects') {
-            $obj_ids = [];
-            if ($this->http->wrapper()->query()->has('search_' . $this->getObjectType())) {
-                $obj_ids = [
-                    $this->http->wrapper()->query()->retrieve(
-                        'search_' . $this->getObjectType(),
-                        $this->refinery->kindlyTo()->int()
-                    )
-                ];
-            } elseif ($this->http->wrapper()->post()->has('search_' . $this->getObjectType())) {
-                $obj_ids = $this->http->wrapper()->post()->retrieve(
-                    'search_' . $this->getObjectType(),
-                    $this->refinery->kindlyTo()->listOf($this->refinery->kindlyTo()->int())
-                );
-            }
-
-            if ($obj_ids !== []) {
-                $this->mailObjects();
-            } else {
-                $this->tpl->setOnScreenMessage('info', $this->lng->txt('mail_select_' . $this->getObjectType()));
-                $this->showMyObjects();
-            }
-        } elseif ($view === $this->getObjectType() . '_members') {
-            $usr_ids = [];
-            if ($this->http->wrapper()->query()->has('search_members')) {
-                $usr_ids = [
-                    $this->http->wrapper()->query()->retrieve(
-                        'search_members',
-                        $this->refinery->kindlyTo()->int()
-                    )
-                ];
-            } elseif ($this->http->wrapper()->post()->has('search_members')) {
-                $usr_ids = $this->http->wrapper()->post()->retrieve(
-                    'search_members',
-                    $this->refinery->kindlyTo()->listOf($this->refinery->kindlyTo()->int())
-                );
-            }
-
-            if ($usr_ids !== []) {
-                $this->mailMembers();
-            } else {
-                $this->tpl->setOnScreenMessage('info', $this->lng->txt('mail_select_one_entry'));
-                $this->showMembers();
-            }
+        $usr_ids = $this->http->wrapper()->query()->has('contact_mailinglist_search_members_ids')
```

Comment:
This is not a `dictOf`, but a `listOf`. Please also try to use `byTring` and `always` to avoid the `has` check.